modules/server/templates/scripts/forseti-server/initialize_forseti_services.sh.tpl

```diff
@@ -0,0 +1,165 @@
+# a caller fails to export the following expected environmental variables, this
+# script will fail immediately rather than partially succeeding.
+echo "Cloud SQL Instance Connection string: ${project_id}:${cloudsql_region}:${cloudsql_instance_name}"
+echo "SQL port: ${cloudsql_db_port}"
+echo "Forseti DB name: ${cloudsql_db_name}"
+
+if ! [[ -f ${forseti_server_conf_path} ]]; then
+    echo "Could not find the configuration file: ${forseti_server_conf_path}." >&2
+    exit 1
+fi
+
+# We had issue creating DB user through deployment template, if the issue is
+# resolved in the future, we should create a forseti db user instead of using
+# root.
+SQL_SERVER_LOCAL_ADDRESS="mysql+pymysql://${cloudsql_db_user}:${cloudsql_db_password}@127.0.0.1:${cloudsql_db_port}"
+FORSETI_SERVICES="explain inventory model scanner notifier"
+
+FORSETI_COMMAND="$(which forseti_server) --endpoint '[::]:50051'"
+FORSETI_COMMAND+=" --forseti_db $SQL_SERVER_LOCAL_ADDRESS/${cloudsql_db_name}?charset=utf8"
+FORSETI_COMMAND+=" --config_file_path ${forseti_server_conf_path}"
+FORSETI_COMMAND+=" --services $FORSETI_SERVICES"
+
+CONFIG_VALIDATOR_COMMAND="${forseti_home}/external-dependencies/config-validator/ConfigValidatorRPCServer"
+CONFIG_VALIDATOR_COMMAND+=" --policyPath='${policy_library_home}/policy-library/policies'"
+CONFIG_VALIDATOR_COMMAND+=" --policyLibraryPath='${policy_library_home}/policy-library/lib'"
+CONFIG_VALIDATOR_COMMAND+=" -port=50052"
+
+if [ "${policy_library_sync_enabled}" == "true" ]; then
+  POLICY_LIBRARY_SYNC_COMMAND="$(which docker) run -d"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --log-driver=gcplogs"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --log-opt gcp-log-cmd=true"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --log-opt labels=git-sync"
+  POLICY_LIBRARY_SYNC_COMMAND+=" -v ${policy_library_home}:/tmp/git"
+  POLICY_LIBRARY_SYNC_COMMAND+=" -v /etc/git-secret:/etc/git-secret"
+  POLICY_LIBRARY_SYNC_COMMAND+=" k8s.gcr.io/git-sync:${policy_library_sync_git_sync_tag}"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --branch=${policy_library_repository_branch}"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --dest=policy-library"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --max-sync-failures=-1"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --repo=${policy_library_repository_url}"
+  POLICY_LIBRARY_SYNC_COMMAND+=" --wait=30"
+
+  # If the SSH file is present, tell git-sync to use SSH to connect to the repo
+  if [ -e "/etc/git-secret/ssh" ]; then
+    POLICY_LIBRARY_SYNC_COMMAND+=" --ssh"
+  fi
+
+  # If the known_hosts file is NOT present, tell git-sync to not check known hosts
+  if ! [ -e "/etc/git-secret/known_hosts" ]; then
+    POLICY_LIBRARY_SYNC_COMMAND+=" --ssh-known-hosts=false"
+  fi
+fi
+
+# Update the permission of the config validator.
+sudo chmod ugo+x ${forseti_home}/external-dependencies/config-validator/ConfigValidatorRPCServer
+
+SQL_PROXY_COMMAND="$(which cloud_sql_proxy)"
+SQL_PROXY_COMMAND+=" -instances=${project_id}:${cloudsql_region}:${cloudsql_instance_name}=tcp:${cloudsql_db_port}"
```

Comment:
You can probably pass in the connection string (everything leading up to the `=`) as a variable.

Reply:
Done